src/component/modal/editRange/forms/components/CouplingsTable.jsx

```diff
@@ -0,0 +1,172 @@
+/** @jsx jsx */
+import { jsx, css } from '@emotion/core';
+import { useFormikContext } from 'formik';
+import { memo, useEffect, useCallback } from 'react';
+import { FaMinus, FaPlus } from 'react-icons/fa';
+
+import Button from '../../../../elements/Button';
+import Input from '../../../../elements/formik/Input';
+import SelectBox from '../../../../elements/formik/SelectBox';
+import { Multiplets } from '../../../../panels/extra/constants/Multiplets';
+import {
+  hasCouplingConstant,
+  translateMultiplet,
+} from '../../../../panels/extra/utilities/MultiplicityUtilities';
+
+const CouplingsTableStyle = css`
+  border-spacing: 0;
+  border: 1px solid #dedede;
+  width: 100%;
+  height: 100%;
+  font-size: 12px;
+
+  th,
+  td {
+    text-align: center;
+    margin: 0;
+    padding: 0.4rem;
+    border-bottom: 1px solid #dedede;
+    border-right: 1px solid #dedede;
+
+    button {
+      background-color: transparent;
+      border: none;
+    }
+
+    .add-button {
+      color: blue;
+    }
+
+    .delete-button {
+      border-radius: 25%;
+      padding: 0;
+      padding-left: 0.3rem;
+      padding-right: 0.3rem;
+    }
+    .delete-button:hover {
+      background-color: red;
+      .icon {
+        color: white;
+      }
+    }
+
+    input {
+      background-color: transparent;
+      border: 0.5px solid #dedede;
+      height: 100%;
+      width: 100%;
+      text-align: center;
+    }
+    .selectBox {
+      background-color: transparent;
+      border: 0.5px solid #dedede;
+      height: 100%;
+      width: 100%;
+      text-align: center;
+    }
+  }
+
+  tr {
+    :last-child {
+      border-top: 3px solid #dedede;
+    }
+  }
+`;
+
+const errorStyle = {
+  border: '1px solid red',
+};
+
+const CouplingsTable = memo(({ push, remove, onFocus, onBlur }) => {
+  const { values, setFieldValue, errors } = useFormikContext();
+
+  useEffect(() => {
+    if (values.signals[values.activeTab]) {
+      values.signals[values.activeTab].j.forEach((_coupling, i) => {
+        if (!hasCouplingConstant(_coupling.multiplicity)) {
+          setFieldValue(`signals.${values.activeTab}.j.${i}.coupling`, '');
+        }
+      });
+    }
+  }, [setFieldValue, values.activeTab, values.signals]);
+
+  const hasMissingCouplingValue = useCallback(
+    (i) =>
+      errors.signals &&
+      errors.signals[`${values.activeTab}`] &&
+      errors.signals[`${values.activeTab}`].missingCouplings.some(
+        (_error) => _error.index === i,
+      ),
```

Comment:
use get from lodash to terminate unnecessary check and && and result [] as default value for get